shortfin/examples/python/mobilenet_server/inference_system.py

```diff
@@ -40,7 +40,12 @@ async def run(self):
             # just writing to the backing storage is the best we have API
             # support for. Generally, APIs on storage should be mirrored onto
             # the array.
-            self.host_staging.storage.data = request.raw_image_data
+            # TODO: Easier to use API for writing into the storage
```

Comment:
Grrg. Yeah, these low level transfer methods need an upgrade. Up until the most recent commit, I didn't have the device level metadata I needed to determine the right way to load data on to the device (i.e. map, transfer, chunk, etc). Now that I have that, need to design a more ergonomic API.

With that said, I believe that `mapping.fill(buffer)` will do what you want if the buffer is the size of the target (i.e. it will copy x1 vs tile). There are some tests that exercise that.

So I think you can say:

```
with self.host_staging.storage.map(write=True, discard=True) as m:
  m.fill(request.raw_image_data)
```

(assuming sizes line up)